EC-Delete-LandingZone.py

```diff
@@ -0,0 +1,529 @@
+            time.sleep(1)
+            response = cfn.list_stack_instances(StackSetName=stackset,Filters=filter)
+
+        try:
+            operationPreferences={
+                'RegionConcurrencyType': 'PARALLEL',
+                'FailureToleranceCount': 9,
+                'MaxConcurrentCount': 10
+            }
+            cfn.delete_stack_instances(
+                StackSetName=stackset, 
+                Regions=regions, 
+                Accounts=[account_id],
+                OperationPreferences=operationPreferences,
+                RetainStacks=False
+                )
+
+            time.sleep(5)
+            response = cfn.list_stack_set_operations(StackSetName=stackset)
+            while(any(x['Status'] == "RUNNING" for x in response['Summaries'])):
+                time.sleep(10)
+                response = cfn.list_stack_set_operations(StackSetName=stackset)
+            
+        
+            print(f"\033[2K\033[1GRemove stacks from StackSet {stackset} [{Status.OK.value}]")
+            
+        
+        except ClientError as error:
+            print(f"\033[2K\033[1GRemove stacks from StackSet {stackset} failed. Reason: {error.response['Error']['Message']} [{Status.FAIL.value}]")
+            print("Exiting...")
+            sys.exit(1)
+
+def delete_stack(stack,session):
+    """
+    Function that updates a stack defined in the parameters
+        :stack:         The stack name
+        :session:       session of the account to be processed
+        :return:        True or False
+    """
+    client = session.client('cloudformation')
+    
+    print(f"Delete {stack} stack ", end="")
+
+    try:
+        describe = client.describe_stacks(StackName=stack)
+        
+        if describe['Stacks'][0]['StackStatus'] not in ('CREATE_COMPLETE', 'UPDATE_COMPLETE', 'ROLLBACK_COMPLETE','UPDATE_ROLLBACK_COMPLETE'):
+            print(f"Cannot delete stack {stack}. Current status is : {describe['Stacks'][0]['StackStatus']} [{Status.FAIL.value}]")
+            print("Exiting...")
+            sys.exit(1)
+
+        client.update_termination_protection(EnableTerminationProtection=False, StackName=stack)
+
+    except ClientError as err:
+        if err.response['Error']['Code'] == 'AmazonCloudFormationException':
+            print(f"\033[2K\033[1GDelete stack {stack} [{Status.NO_ACTION.value}]")
+            return
+        elif err.response['Error']['Code'] == 'ValidationError':
+            print(f"\033[2K\033[1GDelete stack {stack} [{Status.NO_ACTION.value}]")
+            return
+        else:
+            print(f"\033[2K\033[1GDelete stack {stack} failed. Reason : {err.response['Error']['Message']} [{Status.FAIL.value}]")
+            print("Exiting...")
+            sys.exit(1)
+
+    print("in progress ", end="")
+    
+    with Spinner():
+        try:
+            
+            client.delete_stack(StackName=stack)
+            updated=False
+            while updated == False: 
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The `==` and `!=` operators use the compared objects&#39; ` __eq__` method to test if they are equal. To check if an object is a singleton, such as `None`, we recommend that you use the `is` identity comparison operator.

[Learn more](https://docs.python.org/3/reference/expressions.html#is)